cmd/command.go

```diff
@@ -6,8 +6,13 @@ import (
 	"os"
 )
 
+// PreInit defines all the flags that are going to be used.
+// If the default value is not going to be used,
+// the flags will be set as part of the flags.Parse function.
+// In Args, several paths will be changed to be under the updated layers directory (if the user didn't set them using a flag)
+// TODO: should we add more documentation?
 type Command interface {
-	Init()
+	PreInit()
```

Comment:
I agree that `PreInit` isn't the best name since you expect to have an `Init` step after it. But I think that `Init` is also pretty confusing because the flags aren't being initialized as part of this function. If I'm not mistaken, in this function we only define the flags that we're going to use, so maybe `SetNeededFlags` is a better name?

Reply:
Haha I feel like my brain interprets this the opposite way. The command, including the flagset, is being initialized but the values aren't actually _set_ until later. In the original PR it just called `Flags` but we changed it on what I thought was a solid suggestion in the code review https://github.com/buildpacks/lifecycle/pull/232#discussion_r373076090

Reply:
If the `Parse` function was being called inside it, I think it would be very clear to call it `Init`.
What about moving [these lines](https://github.com/buildpacks/lifecycle/blob/b2ce634c61ee84c57d1980c2d80b732311d200c4/cmd/command.go#L28-L38) to be part of the `Init` function?
Then in the `Run` function, we will have:
```
if err := c.Init(asSubcommand); err != nil {
    Exit(err)
}
```

Reply:
After talking with @ekcasey, we agreed on just changing the `Init` function name to `DefineFlags`.